Sources/Core/GTMSessionFetcher.m

```diff
@@ -1964,14 +1966,26 @@ - (void)forgetSessionIdentifierForFetcherWithoutSyncCheck {
   }
 }
 
-// External stop method
+// External stop method and don't call callbacks
 - (void)stopFetching {
+  [self stopFetching:NO];
+}
+
+// External stop method with do call callbacks option
+- (void)stopFetching:(BOOL)shouldCallCallbacks {
   @synchronized(self) {
     GTMSessionMonitorSynchronized(self);
 
     // Prevent enqueued callbacks from executing.
     _userStoppedFetching = YES;
   }  // @synchronized(self)
+  if (shouldCallCallbacks) {
+    NSError *error = [NSError errorWithDomain:kGTMSessionFetcherErrorDomain
+                                   code:GTMSessionFetcherErrorCancelled
+                                     userInfo:@{@"description" : @"Operation cancelled"}];
+    _cancelAndCallback = YES;
```

Comment:
Note that access to `atomic` properties underlying ivars are not implicitly synchronized (ever), and so either need to use `.` property access or be wrapped in the appropriate locking (if the getter/setter are implemented manually).